### ISSUE TITLE
Can't set alpha for some colors

### ISSUE THREAD

**Issue by andreicristianpetcu** [mozilla/FirefoxColor#942]
I have a theme where I set these colors with an alpha.

popup_highlight
sidebar_highlight_text
sidebar_text
toolbar_field_separator
toolbar_field_text_focus
toolbar_top_separator

I can't replicate it with Firefox Color. More context here: https://github.com/mozilla/FirefoxColor/pull/940#issuecomment-662641273

**Comment by andreicristianpetcu** [comment]
I tested all of them and these are the colors that support alpha:

```
export const colorsWithAlpha = [
  "button_background_active",
  "button_background_hover",
  "frame_inactive",
  "frame",
  "icons_attention",
  "icons",
  "ntp_background",
  "ntp_text",
  "popup_border",
  "popup_highlight_text",
  "popup_highlight",
  "popup_text",
  "popup",
  "sidebar_border",
  "sidebar_highlight_text",
  "sidebar_highlight",
  "sidebar_text",
  "tab_line",
  "tab_selected",
  "tab_text",
  "toolbar_bottom_separator",
  "toolbar_field_border_focus",
  "toolbar_field_border",
  "toolbar_field_focus",
  "toolbar_field_highlight_text",
  "toolbar_field_highlight",
  "toolbar_field_separator",
  "toolbar_field_text_focus",
  "toolbar_field_text",
  "toolbar_field",
  "toolbar_text",
  "toolbar_top_separator",
  "toolbar_vertical_separator",
  "toolbar",
];
```

tab_background_separator simply does not work at all on my machine. Not sure if I should add it in the alpha or not.

**Comment by Rob--W** [comment]
What does it mean to not support an alpha?

**Comment by andreicristianpetcu** [comment]
I added all the colors to colorsWithAlpha. I took all of them one by one and made them fully transparent. The ones above took the transparency value. The missing ones stayed opaque regardless of what I did.

**Comment by andreicristianpetcu** [comment]
Also, some if them had a transparency minimum. I could make them transparent but not fully transparent. The popup in the burger menu can be invisible, the megabar can be just partially transparent.

**Comment by Rob--W** [comment]
The Color extension accepted the values though, so I suggest to support alpha for all colors.

**Comment by andreicristianpetcu** [comment]
so shoul I remove colorsWithAlpha?

**Comment by Rob--W** [comment]
It was introduced in #55 (referring to #30), for some reason. I don't know why.

Rather than listing colors that support alpha, are there any colors where it is not supported? To start with, I'd be interested in the set of non-advanced colors.

What happens when alpha isn't supported?

**Comment by andreicristianpetcu** [comment]
I tested both standard and advanced colors. Alpha either works, partially works until a limit or is ignored with no visible side effects. The ones in my list is where it gets ignored, I tested all of them.

**Comment by Rob--W** [comment]
There are currently 38 colors (9 custom colors, 29 advanced colors).
Your list of colors with supported alpha contains 34 colors (1 unknown, 3 "alpha not supported").

* `tab_background_text` ([internally maps to `textcolor`](https://searchfox.org/mozilla-central/rev/227f22acef5c4865503bde9f835452bf38332c8e/toolkit/components/extensions/parent/ext-theme.js#183-184)), for which [alpha is stripped](https://searchfox.org/mozilla-central/rev/227f22acef5c4865503bde9f835452bf38332c8e/toolkit/modules/LightweightThemeConsumer.jsm#45-52).
* `frame` [maps to `accentcolor`](https://searchfox.org/mozilla-central/rev/227f22acef5c4865503bde9f835452bf38332c8e/toolkit/components/extensions/parent/ext-theme.js#177-178), where [alpha is stripped too](https://searchfox.org/mozilla-central/rev/227f22acef5c4865503bde9f835452bf38332c8e/toolkit/modules/LightweightThemeConsumer.jsm#33-38).
* `sidebar` [is just `sidebar`](https://searchfox.org/mozilla-central/rev/227f22acef5c4865503bde9f835452bf38332c8e/toolkit/components/extensions/parent/ext-theme.js#223,230), whose [alpha is dropped too](https://searchfox.org/mozilla-central/rev/227f22acef5c4865503bde9f835452bf38332c8e/browser/modules/ThemeVariableMap.jsm#148-157).

All other properties do appear to support alpha.
Since alpha is supported by default, I suggest to invert the logic. Keep alpha unless it is explicitly listed as not supported.

**Comment by andreicristianpetcu** [comment]
@Rob--W some preset theme colors have `"a": 1` and some don't in the same theme. Either I always add the alpha and the `should not change preset themes` fails since I add it to some colors that don't have it, or I only add it when `a!==1` and then it fails when I remove it. What should I do?

I propose to remove it when alpha===1 and change the preset themes to match.

**Comment by andreicristianpetcu** [comment]
Let's move the conversation on the PR directly https://github.com/mozilla/FirefoxColor/pull/944

**Comment by andreicristianpetcu** [comment]
when should I expect to see this live? @Rob--W

**Comment by Rob--W** [comment]
I just did some quick testing, and noticed that the preview in the page doesn't look like the real theme (e.g. the search bar color and the popup background).

The mismatch in previewed theme and actual is not user-friendly. I'd like to have that fixed before publishing a new release.

**Comment by andreicristianpetcu** [comment]
Search bar with max blue http://localhost:8080/?theme=XQAAAAIGAQAAAAAAAABBKYhm849SCia2CaaEGccwS-xNKliFuyof_iFY4LTdRm0zbNXRVAF7ZvkTn5QQ2-t9nzLQFTpifiMl8GKPfmsqgwdA8IDH2jdY3mS1cmtKqrn0E3OxKHP1EBfWMpiDW0bSNdouHorYHNE_c-m7ImQodaDwmr6_BdGAxTbcUL_I8fEjqF-7PzRHC_LIcWxDDk3NaPPqdPBCid3B5c2UwXas8yDi1kbTH_92HyMA
![image](https://user-images.githubusercontent.com/922368/89075660-62ceff00-d387-11ea-88dd-e8d9efd75b5a.png)

Search bar with blue color but full transparency. http://localhost:8080/?theme=XQAAAAIGAQAAAAAAAABBKYhm849SCia2CaaEGccwS-xNKliFuyof_iFY4LTdRm0zbNXRVAF7ZvkTn5QQ2-t9nzLQFTpifiMl8GKPfmsqgwdA8IDH2jdY3mS1cmtKqrn0E3OxKHP1EBfWMpiDW0bSNdouHqW6JEAonl_iRm22XS3YXknK766aZeR9rqMeXE-79wesdkyeRo8PqfQatmmjWEHoj2MX3f76s68FF5ahSNUanFIHSb__tkPk4A
![image](https://user-images.githubusercontent.com/922368/89075706-7e3a0a00-d387-11ea-883f-e86e77da0814.png)

What is your expectation? I tend to miss small details but for me they look similar @Rob--W

**Comment by andreicristianpetcu** [comment]
Full red popup http://localhost:8080/?theme=XQAAAAIBAQAAAAAAAABBKYhm849SCia2CaaEGccwS-xMDPsqufDOWr6IhCp9RnjnsdP9GNu2SjSQBIXJelHlUb_p3iV0Tk_tp5WixtO7xQnun54TM0zzonAd6w0T64epRIX-MPh_WdfCiLMsTpHEXGfCIspKwaqBlVwgKloTziUxUNFaBxlHf93roKpsyb3_RSU6pROBa9KGdq7F4GlkWH_-mroPnQuOhmu1w3WoEXTtRWEyn__pXnig
![image](https://user-images.githubusercontent.com/922368/89076183-7b8be480-d388-11ea-9541-f1f92f88850f.png)

Transparent red popup http://localhost:8080/?theme=XQAAAAIEAQAAAAAAAABBKYhm849SCia2CaaEGccwS-xMDPsqufDOWr6IhCp9RnjnsdP9GNu2SjSQBIXJelHlUb_p3iV0Tk_tp5WixtO7xQnun54TM0zzonAd6w0T64epRIX-MPh_WdfCiLMsTpHEXGfCIspKwaqBlVwgKloTziUxUNFaBxlHf93roKpstj2AA3M5CdwgC7U-V6AVG9-46JOoTyZX6Ouwlz248f-J2eMgznfCJP_nf__Ku0wA
![image](https://user-images.githubusercontent.com/922368/89076304-aece7380-d388-11ea-9155-4314dd4b1f14.png)
Is this what's bothering you?
![image](https://user-images.githubusercontent.com/922368/89076325-b726ae80-d388-11ea-9ac8-0ce92a1564bc.png)

No focus 
![image](https://user-images.githubusercontent.com/922368/89076358-cad21500-d388-11ea-8771-0e88dbf131a9.png)

**Comment by andreicristianpetcu** [comment]
I guess a clip path with a triangle should do the trick

**Comment by andreicristianpetcu** [comment]
Here is the fix for the popup
![image](https://user-images.githubusercontent.com/922368/89087972-cca6d300-d39e-11ea-83c3-1afb73ebe994.png)
![image](https://user-images.githubusercontent.com/922368/89087976-d29cb400-d39e-11ea-8aeb-ebf37583aeab.png)
![image](https://user-images.githubusercontent.com/922368/89087979-d7616800-d39e-11ea-8079-32d26c3bc241.png)
![image](https://user-images.githubusercontent.com/922368/89087991-de887600-d39e-11ea-893b-8b1dc9a521f2.png)
![image](https://user-images.githubusercontent.com/922368/89087999-e516ed80-d39e-11ea-9e85-be20e5400970.png)
![image](https://user-images.githubusercontent.com/922368/89088003-eb0cce80-d39e-11ea-9403-5db17f22065f.png)

**Comment by andreicristianpetcu** [comment]
![image](https://user-images.githubusercontent.com/922368/89088168-9b7ad280-d39f-11ea-8333-2b889b2e66d5.png)
![image](https://user-images.githubusercontent.com/922368/89088173-9f0e5980-d39f-11ea-9c68-ce5ffde97a33.png)
![image](https://user-images.githubusercontent.com/922368/89088175-a33a7700-d39f-11ea-87b5-2cdb8f7edb2d.png)

**Comment by Rob--W** [comment]
Previously, I tested with add-on version 2.1.6 + the new website (with your change), and although the preview showed transparency, the browser did not. That's what I considered inconsistent and why I referred to it as a bad UX.

But apparently there is nothing wrong with the code itself: When I tested again, but with the add-on built from the source instead of 2.1.6, the feature works as intended. So for this feature to work completely, not only the website should be updated, but the the add-on too. I hoped for this to be a website-only update, but unfortunately it is not.

I'd like to not release too frequent add-on updates - if I publish a new version with your changes, will that fully address your needs, or is there still something else?
